src/util/expr.h

```diff
@@ -192,7 +192,7 @@ class exprt:public irept
     }
     depth_iteratort &operator++()
```

Comment:
Could you please consider doing this without "infinite loop"
maybe like this?

depth_iteratort &operator++()
{
  PRECONDITION(!m_stack.empty());
  while(m_stack.back().it==m_stack.back().end)
  {
    m_stack.pop_back();
    if(m_stack.empty())
    {
      return *this;
    }
    m_stack.back().it++;
  }
  push_expr(*m_stack.back().it);
  return *this;
}